server/src/main/java/org/apache/druid/server/coordinator/duty/RunRules.java

```diff
@@ -112,6 +115,9 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
         if (rule.appliesTo(segment, now)) {
           stats.accumulate(rule.run(coordinator, paramsWithReplicationManager, segment));
           foundMatchingRule = true;
+          if (rule instanceof BroadcastDistributionRule) {
+            broadcastDatasources.add(segment.getDataSource());
```

Comment:
Would you add some comment about that this duty should run before `BalanceSegments` [here](https://github.com/apache/druid/blob/master/server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java#L612)?

Reply:
Added a comment